### PR TITLE
fix composer setup and some failed test

### PR DESCRIPTION
**Pull request by pistej**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes/no
| New feature?  | yes/no
| Breaks BC?    | yes/no
| Tests pass?   | yes/no
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

**Comment by jk2K** [comment]
need fix

File: tests/bootstrap.php [hunk 1]
```diff
@@ -1,15 +1,33 @@
 <?php
+/**
```
Comment (machour) [inline tests/bootstrap.php]: this file can be removed in favor of `./vendor/yiisoft/yii-core/tests/bootstrap.php` (should be updated in phpunit.xml.dist)

Comment (machour) [inline tests/bootstrap.php]: this file can be removed in favor of ./vendor/yiisoft/yii-core/tests/bootstrap.php (should be updated in phpunit.xml.dist)

File: src/Action.php [hunk 2]
```diff
@@ -82,15 +82,15 @@ public function init()
     public function findModel($id)
     {
         if ($this->findModel !== null) {
-            return call_user_func($this->findModel, $id, $this);
+            return \call_user_func($this->findModel, $id, $this);
```
Comment (machour) [inline src/Action.php]: any reason for adding \ for global functions ?

File: tests/bootstrap.php (hunk 1 above)
Comment (pistej) [reply tests/bootstrap.php]: fixed

**Comment by pistej** [comment]
Fixed all tests

File: src/Action.php [hunk 3]
```diff
@@ -8,7 +8,7 @@
 namespace yii\rest;
 
 use Yii;
```
Comment (machour) [inline src/Action.php]: Can be safely removed

File: src/Serializer.php [hunk 4]
```diff
@@ -8,6 +8,7 @@
 namespace yii\rest;
 
 use Yii;
```
Comment (machour) [inline src/Serializer.php]: Here too

File: src/Serializer.php [hunk 5]
```diff
@@ -116,18 +117,33 @@ class Serializer extends Component
...
-    public function init()
+    /*public function init()
     {
         if ($this->request === null) {
             $this->request = Yii::$app->getRequest();
         }
         if ($this->response === null) {
             $this->response = Yii::$app->getResponse();
         }
+    }*/
```
Comment (machour) [inline src/Serializer.php]: Should be removed

File: tests/UrlRuleTest.php [hunk 6]
```diff
@@ -5,11 +5,12 @@
  * @license http://www.yiiframework.com/license/
  */
 
-namespace yiiunit\rest;
+namespace yii\rest\tests\unit;
 
 use Yii;
```
Comment (machour) [inline tests/UrlRuleTest.php]: can be removed

**Comment by pistej** [comment]
Fixed, but there are more usages of `Yii::$app->` and `init()` functions in the code, I'm not sure how to fix it.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**